java/client/src/main/java/glide/api/models/commands/stream/StreamReadGroupOptions.java

```diff
@@ -0,0 +1,72 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.stream;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xreadgroup(Map, String, String,
+ * StreamReadGroupOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/xreadgroup/">redis.io</a>
+ */
+@SuperBuilder
+public final class StreamReadGroupOptions extends StreamReadOptions {
+
+    public static final String READ_GROUP_REDIS_API = "GROUP";
+    public static final String READ_NOACK_REDIS_API = "NOACK";
+
+    /**
+     * If set, messages are not added to the Pending Entries List (PEL). This is equivalent to
+     * acknowledging the message when it is read.
+     */
+    private boolean noack;
+
+    public abstract static class StreamReadGroupOptionsBuilder<
+                    C extends StreamReadGroupOptions, B extends StreamReadGroupOptionsBuilder<C, B>>
+            extends StreamReadOptions.StreamReadOptionsBuilder<C, B> {
```

Comment:
o_O
is it a lombok requirement?

Reply:
yeah... superbuilders make super boilerplate code.  This works... although, I agree with you that it doesn't look pretty.